ryan_savill/02_intro_to_skimage/readme.md

```diff
@@ -0,0 +1,288 @@
+        
+        # this structuring element determines the size of background subtraction
+        str_el = disk(radius)
+        
+        if light_bg:
+            # the tophat function needs the image as well as the structuring element
+            return black_tophat(image, str_el)
+        
+        else:
+            return white_tophat(image, str_el)
+
+# calling the function        
+b_sub_triboium = subtract_background(tribolium,15)        
+
+# thresholding with previous function
+thresh_b_sub_trib = median_and_otsutresh(b_sub_triboium)
+
+# plotting-------------------------------------------------------------
+# This time we want three plots so we need to change the axes and values accordingly!
+fig, (ax1, ax2, ax3) = plt.subplots(1, 3, figsize =(20,20))
+
+ax1.imshow(tribolium)
+ax1.set_title('Original')
+
+ax2.imshow(b_sub_triboium) 
+ax2.set_title('Background Subtracted')
+
+ax3.imshow(thresh_b_sub_trib)
+ax3.set_title('Thresholded Image')
+```
+
+
+![png](output_15_1.png)
+    
+
+
+To actually get to individual marked cells we need to perform connected components analysis, in which each seperate group of white pixels is marked as an individual entity. This is done by assigning a different grey value to each pixel and is easily handled with the measure submodule of scikit image. The result can then be plotted 
+
+
+```python
+from skimage import segmentation, measure # importing the libraries
+
+# performing the binarisation in just one line of code!
+binarised_tribolium = median_and_otsutresh(subtract_background(tribolium,15),1) # saving the binarised image as variable
+
+# calling the labelling function from scikit image
+labelled_tribolium = measure.label(binarised_tribolium)
+
+
+
+fig, (ax1, ax2) = plt.subplots(1, 2, figsize =(20,20)) 
+ax1.imshow(binarised_tribolium)
+ax1.set_title('Thresholded Image')
+ax2.imshow(labelled_tribolium)               
+ax2.set_title('Connected Componets Analysis')
+```
+
+   
+![png](output_17_1.png)
+    
+
+
+As you can see after connected components analysis the different segmented nuclei have differing grey values but sometimes it's hard to see the nuclei or to differentiate them from eachother. To get random colors for each cell we need to generate our own colourmap and to do this we will use numpy! A colourmap is nothing than a 2 dimensional array of shape 256 x 3. The 3 is because each colour is made up of a red, green and blue value. If we want to have a black background we need the first element of the colourmap to be black: 0 red, 0 green and 0 blue. After that we can have random values for each next element. Since randomness in computing is never quite real we can also set a seed for the randomness. This means if we run this code with the same seed on another computer we will get the very same 'randomness'. For some more information on random numbers in the computers you can read about it [here](https://www.sharpsightlabs.com/blog/numpy-random-seed/).
```

Comment:
```suggestion
As you can see after connected components analysis the different segmented nuclei have different grey values. However, sometimes it's hard to see the nuclei or to differentiate them from each other. To get random colors for each cell we need to generate our own colourmap and to do this we will use numpy! A colourmap is nothing else than a 2 dimensional array of shape 256 x 3. The 3 is because each colour is made up of a red, green and blue value. If we want to have a black background we need the first element of the colourmap to be black: 0 red, 0 green and 0 blue. After that, we can have random values for the remainig elements. Since randomness in computing is never fully random we can also set a seed for the randomness. This means if we run this code with the same seed on another computer we will get the very same 'randomness'. For some more information on random numbers in the computers you can read about it [here](https://www.sharpsightlabs.com/blog/numpy-random-seed/).
```
minor wording